deploy-service/common/src/main/java/com/pinterest/deployservice/events/EventBridgePublisher.java

```diff
@@ -29,11 +29,12 @@ public EventBridgePublisher(EventBridgeAsyncClient eventBridgeAsyncClient, Strin
   @Override
   public void publish(BuildBean buildBean, String action) {
 
-    // The branch name is actually "master" in this case. Some legacy CI jobs still use "origin/master",
-    // which is a nonexistent branch. Make this correction so consumers of this event can process
-    // branch info properly.
-    if (StringUtils.equalsIgnoreCase("origin/master", buildBean.getScm_branch())) {
-      buildBean.setScm_branch("master");
+    // Some legacy CI jobs still use remote-tracking branch (with prefix "origin/" added to branch name).
+    // Remove this prefix before publishing.
+    final String originPrefix = "origin/";
+    if (StringUtils.startsWithIgnoreCase(buildBean.getScm_branch(), originPrefix) && !StringUtils.equalsIgnoreCase(buildBean.getScm_branch(), originPrefix)) {
+      final String correctedBranch = buildBean.getScm_branch().substring(originPrefix.length());
+      buildBean.setScm_branch(correctedBranch);
```

Comment:
Will mutating the input buildBean object have any other effects? Like where is it used elsewhere after this function and will that change anything (like [here](https://github.com/pinterest/teletraan/blob/master/deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/resource/Builds.java#L241))

Some might say input to methods should not be mutated.